src/components/modals/ManagePlanDerivationGroupsModal.svelte

```diff
@@ -0,0 +1,414 @@
+      },
+      {
+        cellDataType: 'boolean',
+        editable: true,
+        headerName: 'Included in Plan',
+        resizable: false,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        valueGetter: (params: ValueGetterParams<DerivationGroup>) => {
+          const { data } = params;
+          if (data) {
+            return $selectedPlanDerivationGroupNames.includes(data.name);
+          }
+          return false;
+        },
+        width: 115,
+      },
+      {
+        cellClass: 'action-cell-container',
+        cellRenderer: (params: DerivationGroupCellRendererParams) => {
+          const actionsDiv = document.createElement('div');
+          actionsDiv.className = 'actions-cell';
+          new DataGridActions({
+            props: {
+              rowData: params.data,
+              viewCallback: params.viewDerivationGroup,
+              viewTooltip: {
+                content: 'View Derivation Group',
+                placement: 'bottom',
+              },
+            },
+            target: actionsDiv,
+          });
+
+          return actionsDiv;
+        },
+        cellRendererParams: {
+          viewDerivationGroup,
+        } as CellRendererParams,
+        headerName: '',
+        resizable: false,
+        sortable: false,
+        suppressAutoSize: true,
+        suppressSizeToFit: true,
+        width: 40,
+      },
+    ];
+  }
+
+  function viewDerivationGroup(derivationGroup: DerivationGroup) {
+    const dg = $derivationGroups.find(dg => dg.name === derivationGroup.name);
+    selectedDerivationGroup = dg;
+  }
+
+  function getRowId(derivationGroup: DerivationGroup): string {
+    return `${derivationGroup.name}:${derivationGroup.source_type_name}`;
+  }
+
+  function handleUpdateLayerNewDerivationGroups(newLayers: Layer[]) {
+    // Derivation Group association is done on a *plan* level which makes this a little tricky. We want to update *all* available External Event Layers with the new layers (post-association), so we must look through all available timelines & their rows to find the layers we're updating
+    timelines.forEach(currentTimeline => {
+      currentTimeline.rows.forEach(currentRow => {
+        const regeneratedLayers = currentRow.layers.map(currentLayer => {
+          const layerHasNewVersion = newLayers.find(nl => nl.id === currentLayer.id);
+          return layerHasNewVersion !== undefined ? layerHasNewVersion : currentLayer;
+        });
+        viewUpdateRow('layers', regeneratedLayers, currentTimeline.id, currentRow.id);
+      });
+    });
+  }
+
+  function changeDerivationGroupAssociation(checked: boolean, derivationGroupName: string | undefined) {
+    if (derivationGroupName !== undefined) {
+      if (checked) {
+        // insert
+        effects.insertDerivationGroupForPlan(derivationGroupName, $plan, user);
+        if ($derivationGroupPlanLinkError !== null) {
+          showFailureToast('Failed to link derivation group & plan.');
```

Comment:
Can you move the error handling into the effect itself to be consistent with how we show failure toasts?